src/test/java/com/codegym/jrugotom5/service/AdvertServiceTest.java

```diff
@@ -105,31 +105,31 @@ void testGetAdvertsByCreateBy_WithAdverts_ReturnsDtoList() {
         Advert advert2 = new Advert();
         List<Advert> adverts = List.of(advert1, advert2);
 
-        when(advertRepository.getAdvertsByCreatedById(userId)).thenReturn(adverts);
-
-        AdvertInfoForCreatorDto dto1 = new AdvertInfoForCreatorDto();
-        AdvertInfoForCreatorDto dto2 = new AdvertInfoForCreatorDto();
+        List<AdvertInfoForCreatorDto> expected = new ArrayList<>();
 
-        when(modelMapper.map(advert1, AdvertInfoForCreatorDto.class)).thenReturn(dto1);
-        when(modelMapper.map(advert2, AdvertInfoForCreatorDto.class)).thenReturn(dto2);
+        when(advertRepository.getAdvertsByCreatedById(userId)).thenReturn(adverts);
 
+        when(modelMapper.map(any(Advert.class), eq(AdvertInfoForCreatorDto.class)))
+                .thenAnswer(invocation -> {
+                    AdvertInfoForCreatorDto dto = new AdvertInfoForCreatorDto();
+                    expected.add(dto);
+                    return dto;
+                });
         List<AdvertInfoForCreatorDto> result = advertService.getAdvertsByUserId(userId);
 
-        assertEquals(2, result.size());
-        assertTrue(result.contains(dto1));
-        assertTrue(result.contains(dto2));
+        assertEquals(expected, result);
         verify(advertRepository).getAdvertsByCreatedById(userId);
     }
 
     @Test
-    void testGetAdvertsByCreateBy_NoAdverts_ReturnsEmptyList() {
-        Long userId = 1L;
-        when(advertRepository.getAdvertsByCreatedById(userId)).thenReturn(Collections.emptyList());
+    void testGetAdvertsByCreateBy_InvalidUserId_ReturnsException() {
+        Long invalidUserId = 0L;
+        AdvertService advertService = new AdvertService(advertRepository, modelMapper);
```

Comment:
Why did you use it?
Mocks already injected to AdvertService in this test.
See  `public void setUp()` method

Reply:
> Why did you use it? Mocks already injected to AdvertService in this test. See `public void setUp()` method

forgot about setup... 😬😬